tests/helics/application_api/application-api-tests.cpp

```diff
@@ -23,6 +29,15 @@ struct globalTestConfig
     globalTestConfig () = default;
     ~globalTestConfig ()
     {
+        // macOS ZeroMQ can have issues shutting down, if not built from source without CURVE
+#ifdef ENABLE_ZMQ_CORE
+#ifdef __APPLE__
+        if (ZmqContextManager::setContextToLeakOnDelete ())
+        {
+            ZmqContextManager::getContext ().close ();
+        }
+#endif
+#endif
```

Comment:
Just a note that  I am pretty sure this has to do with the fact that Travis is two cores, and apple does some different things with library terminations, so If the context is terminated at the user level, then the library is deleted that never gets propagated to the zmq threads and they try to shut down after the master lock has been deleted resulting in some chaos during the termination sequence.